### PR TITLE
feat(openchallenges): add new challenge category facets

### LINKED ISSUE #2397: [Feat] Add "Continuous benchmarking" as a category

### PR DESCRIPTION
**Pull request by vpchung**
Fixes #2397 

## Changelog
* add two enum values to OpenAPI spec: `benchmark`, `hackathon`
* update SQL table creation to include new values
* update SQL table data insertion to populate data via dump file
* add dump file for challenge_categories
* add "Continuous benchmark" and "Hackathon" to the UI

## Preview
**SQL table**
Screenshot 2023-12-13 at 9 31 40 AM

**Challenge search page**
Screenshot 2023-12-13 at 9 19 59 AM

### CODE REVIEW

**Comment by vpchung** [comment]
@tschaffter do you think this is something we can add to the sprint this month, now that we have ~12 hackathons in our db?

**Comment by vpchung** [comment]
Data has been updated in this branch to reflect the latest hackathons that were added last week:

Screenshot 2024-02-07 at 12 34 09 PM

As noted in the [ticket](https://github.com/Sage-Bionetworks/sage-monorepo/issues/2397#issuecomment-1854418586), I was able to expose the new facets into the UI but I'm not able to get the facets to actually work:

Screenshot 2024-02-07 at 12 34 27 PM

@rrchai @tschaffter would you be able to help with that? Thanks!!

**Comment by tschaffter** [comment]
Property name should use the singular form in the data model. Currently we have a mix of singular and plural.

E.g. here is the header of `incentives.csv`:

```
"","incentives","challenge_id","created_at"
```

**Comment by tschaffter** [comment]
## Update 2023-02-07

The Challenge service can now filter benchmarks and hackathons.

```
GET {{basePath}}/challenges?categories=hackathon

{
  "number": 0,
  "size": 100,
  "totalElements": 12,
  "totalPages": 1,
  "hasNext": false,
  "hasPrevious": false,
  "challenges": [
    {
      "id": 497,
      "slug": "hack-rare-disease",
      "name": "Harvard Rare Disease Hackathon 2024",
      "headline": "Are you a student interested in using AI/ML to tackle rare diseases? Join us!",
      "description": "This March 2-3, join us for the 2024 Harvard Rare Disease Hackathon, where students will gather on Harvard’s campus to set forth their own data-driven solutions for rare diseases.  Participants will have the chance to analyze public and patient-sourced genomic and clinical datasets, and will be challenged to produce deliverables for participating patient organizations. These deliverables may take the form of a data report, computational tool, or web/mobile application that improves the lives of patients or furthers research progress.   Participation is free and open to all undergraduate and graduate students who register with their .edu email address.",
      "doi": "",
      "status": "upcoming",
      "platform": {
        "id": null,
        "slug": null,
        "name": null
      },
      "websiteUrl": "https://www.harvard-rarediseases.org/",
      "avatarUrl": "",
      "incentives": [
        "monetary"
      ],
      "submissionTypes": [
        "prediction_file",
        "container_image",
        "notebook"
      ],
      "inputDataTypes": [],
      "categories": [
        "hackathon"
      ],
      "startDate": "2024-03-02",
      "endDate": "2024-03-03",
      "starredCount": 0,
      "createdAt": "2024-02-06T00:12:34Z",
      "updatedAt": "2024-02-06T00:41:24Z"
    },
...
```

**Comment by tschaffter** [comment]
## Update 2024-02-12

Here are the changes I made:

- renamed a column of the SQL table that tracks the category to be consistent with the existing naming convention used in other tables: `category` -> `name`
- added the property `Challenge.categories` to the API description
- updated the challenge service
- updated the API client for Angular
- renamed "Continuous Benchmark" to "Benchmark" in the challenge search page as an attempt to simplify (pending team approval)

File: apps/openchallenges/challenge-service/src/main/resources/openapi.yaml [hunk 1]
```diff
@@ -710,6 +712,9 @@ components:
         websiteUrl: https://openchallenges.io
         name: name
         id: 1
+        categories:
+        - featured
+        - featured
```
Comment (tschaffter) [inline apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: Double featured?

**Review by tschaffter — APPROVED** [review]

File: apps/openchallenges/challenge-service/src/main/resources/openapi.yaml (hunk 1 above)
Comment (tschaffter) [reply apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: The file `[apps/openchallenges/challenge-service/src/main/resources/openapi.yaml](https://github.com/Sage-Bionetworks/sage-monorepo/pull/2404/files/2e7a5ac59f7656aacc00a19ee319bdff6ef65694#diff-76fa348d73b49d4f8047ca61c668bcee8289016590ae155948494163883fd457)` is automatically generated so we don't want to modify it manually.

Comment (vpchung) [reply apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: Whoa, nice catch.  Not sure how this happened, as `nx build` was used to generate the API spec.  Will try to re-generate to see if this gets fixed.

Comment (tschaffter) [reply apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: It's the example that OpenAPI automatically generate. We could write our own example but that's a detail.

Comment (vpchung) [reply apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: 🙃 

Screenshot 2024-02-13 at 9 43 25 AM

**Review by gaiaandreoletti — APPROVED** [review]

File: apps/openchallenges/challenge-service/src/main/resources/openapi.yaml (hunk 1 above)
Comment (vpchung) [reply apps/openchallenges/challenge-service/src/main/resources/openapi.yaml:717]: I'm wondering if this is perhaps the result of a merging conflict went awry 🙃

**Pull request merged**